src/Tests/RedirectUITest.php

```diff
@@ -386,4 +379,35 @@ public function assertRedirect($path, $expected_ending_url, $expected_ending_sta
     $this->assertEqual($headers[0][':status'], $expected_ending_status);
   }
 
+  /**
+   * Test cache tags.
+   *
+   * @todo Not sure this belongs in a UI test, but a full web test is needed.
+   */
+  public function testCacheTags() {
+    // Enable internal page cache.
+    $this->container->get('config.factory')->getEditable('system.performance')
+      ->set('cache.page.use_internal', TRUE)
+      ->set('cache.page.max_age', 300)
+      ->save();
+
+    /** @var \Drupal\redirect\Entity\Redirect $redirect1 */
+    $redirect1 = $this->storage->create();
+    $redirect1->setSource('test-redirect');
+    $redirect1->setRedirect('node');
+    $redirect1->setStatusCode(301);
+    $redirect1->save();
+
+    $this->assertRedirect('test-redirect', 'node');
+    $headers = $this->drupalGetHeaders(TRUE);
+    // Note, self::assertCacheTag() cannot be used here since it only looks at
+    // the final set of headers.
+    $this->assertEqual(implode(' ', $redirect1->getCacheTags()), $headers[0]['x-drupal-cache-tags'], 'Redirect cache tags properly set.');
```

Comment:
We should also see a X-Drupal-Cache MISS header here, and when you repeat the redirect, you should get a X-Drupal-Cache HIT.